main/src/hadd.cxx

```diff
@@ -388,9 +389,12 @@ int main( int argc, char **argv )
             }
             std::string line;
             while( indirect_file ){
-               if( std::getline(indirect_file, line) && line.length() ) {
-                  firstInput = TFile::Open(line.c_str());
-                  break;
+               if( std::getline(indirect_file, line) && line.length() && TString(line).EndsWith(".root") &&
+                   gSystem->AccessPathName(line.c_str(), kReadPermission) == kFALSE) {
```

Comment:
Seeing a file and not being able to read it should trigger an error  rather than being ignored silently.

Reply:
The error appears later in the code, so I think it's not really needed at this stage?

See example below:
```
Parallelizing  with 10 processes.
hadd Target file: mkShapes__RDF_2018_v9_emu_ttHMVA.root
hadd compression setting for all output: 101
Each process should handle at least 3 files for efficiency. Setting the number of processes to: 2
hadd Source file 1: mkShapes__RDF_2018_v9_emu_ttHMVA__ALL__top_0.root
hadd Source file 2: mkShapes__RDF_2018_v9_emu_ttHMVA__ALL__top_1.root
hadd Source file 3: mkShapes__RDF_2018_v9_emu_ttHMVA__ALL__top_2.root
Error in <TFile::TFile>: file /tmp/mkShapes__RDF_2018_v9_emu_ttHMVA__ALL__top_2.root does not exist
Error in <TFileMerger::AddFile>: cannot open file mkShapes__RDF_2018_v9_emu_ttHMVA__ALL__top_2.root
hadd failed at the parallel stage
```

Reply:
So the current code count only the valid files to determine the number of cores and the proceed to still attempt to process those files.  This is probably the right thing but require comments in the code that explain this is intentional.

Reply:
Done!